test/ResultSet/ResultSetIntegrationTest.php

```diff
@@ -90,6 +90,20 @@ public function testCanProvideIteratorAsDataSource()
         $this->assertSame($it, $this->resultSet->getDataSource());
     }
 
+    public function testCanProvideArrayAsDataSource()
+    {
+        $dataSource = [['foo']];
+        $this->resultSet->initialize($dataSource);
+        $this->assertEquals($dataSource[0], (array) $this->resultSet->current());
+
+        $returnType = new ReturnType();
+        $dataSource = [$returnType];
+        $this->resultSet->setArrayObjectPrototype($returnType);
```

Comment:
I think it can use : 

```php
$dataSource = [new ArrayObject([], ArrayObject::ARRAY_AS_PROPS)];
$this->resultSet->setArrayObjectPrototype($dataSource);
```
so no need to have `ReturnType.php` file.